### PR TITLE
Splitting projects

### PR DESCRIPTION
**Pull request by andrey-helldar**
> Note
>
> The development is in progress. The structure of localization and script files can change during development.
> Once it is over, everyone will be able to suggest their own adjustments.

https://github.com/Laravel-Lang/lang/discussions/1702#discussioncomment-703215

### CODE REVIEW

**Comment by luisprmat** [comment]
Hopefully the keys starting with `Lang::get('...')` are not forgotten.

For example [these](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Auth/Notifications/ResetPassword.php#L88).

Some translations also begin with [`@lang(...)`](https://github.com/laravel/framework/blob/8.x/src/Illuminate/Notifications/resources/views/email.blade.php#L7)

**Comment by andrey-helldar** [comment]
@luisprmat, thank you!

**Comment by andrey-helldar** [comment]
@caouecs, my code is ready **BUT**:

1. I will remove the Draft tag from this PR when https://github.com/Laravel-Lang/lang/pull/1747 is accepted. I did not add split localization files to this PR, because there is one open PR in the project.
2. It will need to be unloaded only after accepting other PRs. This must be done because I will need to run 1 script file once, which will split the keys. Otherwise, other developers will be forced to manually fix conflicts.
3. Fixed compilation error for status files. [This one](https://github.com/Laravel-Lang/lang/blob/master/app/main/Processors/Statuses/Processor.php#L68-L70) condition dropped the array.
4. The Jetstream and Fortify projects do not have language files, but they use translations. I added these packages in dependencies and when the Foo script is executed, it collects all the keys from them, forming files.
5. I split the files by project:

image

image

image

image

**Comment by andrey-helldar** [comment]
Tests...)))

**Comment by andrey-helldar** [comment]
Tests are ready

image

**Comment by andrey-helldar** [comment]
@caouecs, ready :)

**Comment by andrey-helldar** [comment]
image

**Comment by luisprmat** [comment]
I am seeing as if it was necessary to translate some keys that were already correct?

**Comment by andrey-helldar** [comment]
I'll check again.Best Regards    Andrey Helldar17:00, 16 мая 2021 г., Luis Parrado ***@***.***>:
I am seeing as if it was necessary to translate some keys that were already correct?

File: docs/statuses/es.md [hunk 1]
```diff
@@ -2,7 +2,217 @@
...
+
+##### Missing: 20
+
+<table >
+<tr><td align="left" >
+Add a new team member to your team
+</td>
+</tr>
+<tr><td align="left" >
+Are you sure you want to delete this team? Once a team is deleted
```
Comment (luisprmat) [inline docs/statuses/es.md:48]: [This line for example](https://github.com/Laravel-Lang/lang/blob/master/locales/es/es.json#L56) is correctly translated but appears as missing and also appears incomplete. And there are many more cases

image

**Review by luisprmat — CHANGES_REQUESTED** [review]

File: docs/statuses/es.md (hunk 1 above)
Comment (luisprmat) [reply docs/statuses/es.md:48]: None of the above translations correspond to valid translations

Comment (andrey-helldar) [reply docs/statuses/es.md:48]: Got it. I'll do it. Thank you!

Comment (andrey-helldar) [reply docs/statuses/es.md:48]: @luisprmat, hmm... Regularity is not working properly. I'll fix it.

image

image

image

**Comment by andrey-helldar** [comment]
I found the bug. In the near future I will fix it and inform you.

File: docs/statuses/es.md (hunk 1 above)
Comment (andrey-helldar) [reply docs/statuses/es.md:48]: I found the bug. In the near future I will fix it and inform you.

**Review by luisprmat — APPROVED** [review]
At the moment there are no visible errors regarding the division of the packages. However if a user decides to install let's say Laravel + jetstream they need a way to re-integrate in the `xx.json` file the  jetstream + fortify translations. I assume a publisher ([Yours?](https://github.com/andrey-helldar/laravel-lang-publisher)) will be recommended or embed the files manually? It would have to be said in the documentation ...

**Comment by andrey-helldar** [comment]
@luisprmat, Yes, if you only use files from this project, then developers will need to manually merge the files.

If you use a translation manager, they will combine the keys themselves.

I am currently working on this - if an installation is detected, for example, jetstream and spark, then the manager will automatically save the keys for laravel + jetstream + fortify + spark to the * .json file (jetstream loads fortify during installation).

**Comment by andrey-helldar** [comment]
@Laravel-Lang/laravel-lang, guys, check this PR. Maybe I missed something.

**Comment by caouecs** [comment]
I will try to review the code this week-end. But I have a question : it's for a new version ( v10 ) of this package ? :)

**Comment by andrey-helldar** [comment]
> I will try to review the code this week-end. But I have a question : it's for a new version ( v10 ) of this package ? :)

This point needs to be pondered.

Yes: I took the keys outside.
No: The structure of the source files is intact. I just added new files.

[](https://api.gh-polls.com/poll/01F5ZANKAMDATC6QTVEERFC49H/Keep%20these%20changes%20in%20version%209.x/vote)
[](https://api.gh-polls.com/poll/01F5ZANKAMDATC6QTVEERFC49H/Publish%20these%20changes%20in%20version%2010.x/vote)

**Comment by andrey-helldar** [comment]
> I will try to review the code this week-end. But I have a question : it's for a new version ( v10 ) of this package ? :)

I think it is better to publish these changes in the major version, since a lot of keys have been taken out.

If you save the changes in version 9, then the developers will catch a "bug" - they will lose 662 translated strings.

**Before:**
image

**After:**
image

**Review by dinhquochan — APPROVED** [review]
Great!

**Comment by andrey-helldar** [comment]
That's what I was talking about :)

image

image

**Review by masterix21 — APPROVED** [review]

**Review by masterix21 — APPROVED** [review]

**Review by mohamedsabil83 — APPROVED** [review]
It seems ok for me

**Pull request merged**